src/monty/io.py

```diff
@@ -54,7 +55,53 @@ def zopen(filename: Union[str, Path], *args, **kwargs) -> IO:
     return open(filename, *args, **kwargs)
 
 
-def reverse_readfile(filename: Union[str, Path]) -> Generator[str, str, None]:
+def _get_line_ending(
```

Comment:
This function has a fundamental weakness, such that I might refrain from using it by default, that is: if the last line doesn't end with a newline character, it would not be possible for it (or any function) to determine its line ending. 

As per [the POSIX standard definition of a line](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206):
> A sequence of zero or more non- <newline> characters plus a terminating <newline> character.

For example if we have the following non-standard file (`repr` as `"line1\nline2"`):
```
line1
line2
```
It would be impossible to find what line ending is used.

As opposed to `"line1\nline2\n"` which would look like (note the "empty line" at the end, technically the last empty line is not a line):
```
line1
line2

```

Reply:
> It would be impossible to find what line ending is used.

can you elaborate on this? why is it impossible?

Reply:
Sorry I didn't intend to make this info obfuscatory (these were written as a note while coding, so might lack detailed explanation). Also **I could be wrong here** because I believe this is already violating the POSIX standard and **discussion on this might not happen / be meaningful in the real world**.

Say if a file complies with POSIX standard (every line has an ending, especially the last), its representation could only ends with two possible the line ending (`"\n"` or `"\r\n"`, [there are more possible line endings](https://en.wikipedia.org/wiki/Newline), but we only consider these two here).

If not, someone could write:
```
line1\r
line2
```
in Linux whose `repr` would be `line1\r\nline2` but we can never know what's it intended to be? Ideally those slash should be skipped as `\\r` but one could also "force-write" them in binary mode.

Say if I write the following files on my mac:
<img width="247" alt="image" src="https://github.com/user-attachments/assets/44d1a583-3e57-47eb-a247-ed604ecaf928">

With the following code:
```python
with open("file.txt", mode="rb") as file:
    for line in file:
        print(repr(line.decode()))

```
I ended up getting:
```
'line1\\r\r\n'
'line2'
```

I'm already confused with all these crazy cases myself, so perhaps just throw an error if the file is not complying with POSIX standard, instead of trying to guess what's the line ending :)